Add Helm instructions for updating a license file

ptgott opened the pull request:
Closes #27772

Include this information in `license.mdx`, where we include other information about license renewals and updates.

ptgott committed:
Add Helm instructions for updating a license file

Closes #27772

Include this information in `license.mdx`, where we include other
information about license renewals and updates.

ptgott commented:
@hugoShaka I haven't tested out the instructions—are they correct? Anything else that readers should know? Thanks!

hugoShaka commented on docs/pages/admin-guides/deploy-a-cluster/license.mdx:
````diff
@@ -115,6 +115,34 @@ The Teleport Auth Service checks the expiration status of your license every
+   secret is called `license`, but there may be a different secret name
+   specified in the `licenseSecretName` field of the chart's values file.
+
+   ```code
+   $ kubectl delete secret license
+   ```
+
+1. Create a secret with the same name as the one you deleted:
+
+   ```code
+   $ kubectl create secret generic license --from-file license.pem
+   ```
````
Instead of deleting/re-creating the secret, it's safer to replace it. This way the auth pods are always reboot-proof.

```
$ kubectl create secret generic license --from-file license.pem --dry-run=client -o yaml | kubectl replace -n <namespace> -f -
```

hugoShaka commented on docs/pages/admin-guides/deploy-a-cluster/license.mdx:
```diff
@@ -115,6 +115,34 @@ The Teleport Auth Service checks the expiration status of your license every
 hour. This means that, after you replace your license file and restart your Auth
 Service instances, it can take up to an hour to confirm the new license.
 
+If you are running a self-hosted Teleport Enterprise cluster on Kubernetes using
+the `teleport-cluster` Helm chart, complete the following steps to update your
+license:
+
+1. Download a new license file from your Teleport account using the instructions
+   we provide [earlier in this guide](#download-your-license-file). We assume
+   that the license file is called `license.pem`.
+
+1. Delete the secret that stores your existing license. We assume that the
+   secret is called `license`, but there may be a different secret name
+   specified in the `licenseSecretName` field of the chart's values file.
```
The instructions are missing the namespace. This is a very common mistake our customers are doing: 
 they deploy Teleport in the `teleport` namespace, update the license in the `default` namespace, the license expires and they open a support ticket.

ptgott committed:
Respond to hugoShaka feedback

- Show how to replace the license secret, not destroy it and recreate it
- Instruct the user to specify a namespace

ptgott commented:
@mmcallister Checking if you'd have time to take a look at this one. Thanks!

ptgott committed:
Respond to zmb3 feedback

Add some explanation to the command for replacing a license file in a
Kubernetes-deployed Auth Service instance.

hugoShaka reviewed (approved):
Should we use the namespace var in line 74 as well, when we tell users to create the secret?

I also re-tried the latest instructions and they work:
```bash
$ kubectl -n teleport create secret generic license --from-file=/var/lib/teleport/license.pem
secret/license created

# Editing the secret to observe that the license got replaced
$ kubectl edit -n teleport secret/license
secret/license edited

$ kubectl create secret generic license --from-file /var/lib/teleport/license.pem --dry-run=client -o yaml | kubectl replace -n teleport -f -
secret/license replaced
```
```

ptgott commented:
Good catch, updated!
